src/web/aws-cloudwatch/StepAuthorize.jsx

```diff
@@ -23,57 +24,60 @@ const StepAuthorize = ({ onChange, onSubmit }) => {
           <DisappearingInput id="password" type="password" />
           {/* Continue on, Nothing to See Here */}
 
-          <Input id="awsCloudWatchName"
-                 type="text"
-                 value={formData.awsCloudWatchName ? formData.awsCloudWatchName.value : ''}
-                 onChange={onChange}
-                 placeholder="CloudWatch Integration Name"
-                 label="Name of integration"
-                 autoComplete="off"
-                 required />
+          <ValidatedInput id="awsCloudWatchName"
+                          type="text"
+                          fieldData={formData.awsCloudWatchName}
+                          onChange={onChange}
+                          placeholder="CloudWatch Integration Name"
+                          label="Name of integration"
+                          autoComplete="off"
+                          required />
 
-          <Input id="awsCloudWatchDescription"
-                 type="textarea"
-                 label="Integration description"
-                 placeholder="CloudWatch Integration Description"
-                 onChange={onChange}
-                 value={formData.awsCloudWatchDescription ? formData.awsCloudWatchDescription.value : ''}
-                 rows={4} />
+          <ValidatedInput id="awsCloudWatchDescription"
+                          type="textarea"
+                          label="Integration description"
+                          placeholder="CloudWatch Integration Description"
+                          onChange={onChange}
+                          fieldData={formData.awsCloudWatchDescription}
+                          rows={4} />
 
-          <Input id="awsCloudWatchAwsKey"
-                 type="password"
-                 label="AWS Key"
-                 placeholder="CloudWatch Integration AWS Key"
-                 onChange={onChange}
-                 value={formData.awsCloudWatchAwsKey ? formData.awsCloudWatchAwsKey.value : ''}
-                 autoComplete="off"
-                 pattern="AK[A-Z0-9]{18}"
-                 minLength="20"
-                 required />
+          <ValidatedInput id="awsCloudWatchAwsKey"
```

Comment:
I think the validations for the AWS Key and AWS Secret here are too specific (checking that exactly 20/40 characters in length and a key starting with "AK"). I definitely see that from several key/secret pair examples, this seems to be true. However, the [official documentation (Access Keys section)](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html) does not indicate that this will always be the case. What if a key or secret is generated in the future that does not exactly match these checks? I just want to make sure we don't have a blocker issue come in at a later date due to a user not being able to enter their key/secret.

I think we should only check whether they have entered a non-blank value for the Key/Secret. The helper text below the field indicating the length/pattern expectations should be enough to tell the user what they're looking to enter. We'll fully check the key/secret anyhow when performing other API calls. 

Looping in @ceruleancee for input and reference too.

Reply:
_Carry over from PMs for posterity._

Claudia and I discussed the Key and Secret patterns and went with a simplified version of: https://aws.amazon.com/blogs/security/a-safer-way-to-distribute-aws-credentials-to-ec2/